x-pack/packages/kbn-ai-assistant/src/chat/chat_body.tsx

```diff
@@ -226,9 +226,36 @@ export function ChatBody({
   });
 
   const handleCopyConversation = () => {
+    const deserializeMessage = (message: Message): Message => {
+      const copiedMessage = cloneDeep(message);
+
+      const { function_call: functionCall, content, data } = copiedMessage.message;
+
+      if (
+        copiedMessage.message.function_call?.arguments &&
+        typeof copiedMessage.message.function_call.arguments === 'string'
+      ) {
+        copiedMessage.message.function_call.arguments = JSON.parse(functionCall?.arguments ?? '{}');
```

Comment:
Also `JSON.parse` can throw if an invalid json is passed.

Reply:
Resolved